scripts/generate-hashes-from-sources

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# Script for extracting BSD-style checksum into
+# 'file'.'method of checksum' with 'hash' content
+
+set -e
+if [ "${VERBOSE:-0}" -ge 2 ] || [ "${DEBUG:-0}" -eq 1 ]; then
+    set -x
+fi
+
+localdir="$PWD"
+scriptdir="$(dirname "$(readlink -f "$0")")"
+fedora_sources="$1"
+
+# shellcheck source=scripts/functions.sh
+source "$scriptdir/functions.sh"
+
+if exist_url "$fedora_sources"; then
+    tmpsources="$(mktemp)"
+    wget -q "$fedora_sources" -O "$tmpsources"
+
+    while read -r src
+    do
+        method="$(echo "$src" | sed -ne 's/^\(.*\) (\(.*\)) = \(.*\)/\1/p')"
+        file="$(echo "$src" | sed -ne 's/^\(.*\) (\(.*\)) = \(.*\)/\2/p')"
+        hash="$(echo "$src" | sed -ne 's/^\(.*\) (\(.*\)) = \(.*\)/\3/p')"
```

Comment:
According to our [coding guidelines](https://www.qubes-os.org/doc/coding-style/#security-coding-guidelines), please name those variable (and `src` too) with `untrusted_` prefix.